addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxPlayerHandler.java

```diff
@@ -396,20 +396,28 @@ public void coverArtChangeEvent(String mac, String coverArtUrl) {
      * @return A RawType object containing the image, null if the content type could not be found or the content type is
      *         not an image.
      */
-    private RawType downloadImage(String url) {
-        if (StringUtils.isNotEmpty(url)) {
-            if (!IMAGE_CACHE.containsKey(url)) {
-                IMAGE_CACHE.put(url, () -> {
-                    logger.debug("Trying to download the content of URL {}", url);
-                    return HttpUtil.downloadImage(url);
-                });
-            }
-            RawType image = IMAGE_CACHE.get(url);
-            if (image == null) {
-                logger.debug("Failed to download the content of URL {}", url);
-                return null;
-            } else {
-                return image;
+    private RawType downloadImage(String mac, String url) {
+        // Only get the image if this is my PlayerHandler instance
+        if (isMe(mac)) {
+            if (StringUtils.isNotEmpty(url)) {
+                if (!IMAGE_CACHE.containsKey(url)) {
```

Comment:
You could consider switching to `IMAGE_CACHE.putIfAbsentAndGet` I added it for the combination of contains, adding if not contains and calling get on it.

Reply:
Good idea.

Reply:
Given that the item being added to the cache is an image, wouldn't this incur a good bit more overhead than the current approach, where the image item is downloaded and placed in the cache only if it is not already in the cache?

Reply:
But as far as I remember it should only do something new if its is absent, so you have any reason to assume why it works differently?

Reply:
Ah, I see now.